progressStats.js

```diff
@@ -0,0 +1,76 @@
+class ProgressStats {
+
+    constructor(totalDownloads) {
+        this.totalDownloads = totalDownloads
+        this.downloadCounter = 0
```

Comment:
Please rename the `totalDownloads` and `downloadCounter` to `totalTasks` and `taskCounter`.